ethcore/types/src/account_diff.rs

```diff
@@ -104,12 +105,9 @@ impl AccountDiff {
 
 // TODO: refactor into something nicer.
 fn interpreted_hash(u: &H256) -> String {
-	if u <= &H256::from(0xffffffff) {
-		format!("{} = 0x{:x}", U256::from(&**u).low_u32(), U256::from(&**u).low_u32())
-	} else if u <= &H256::from(u64::max_value()) {
-		format!("{} = 0x{:x}", U256::from(&**u).low_u64(), U256::from(&**u).low_u64())
-//	} else if u <= &H256::from("0xffffffffffffffffffffffffffffffffffffffff") {
-//		format!("@{}", Address::from(u))
+	let uint = u.into_uint();
+	if let Ok(n) = u64::try_from(uint) {
+		format!("{} = {:#x}", n, n)
```

Comment:
Does `Display` for `H256` print a numeric (decimal) value? If yes, I don't think it makes much sense, cause `H256` is not meant to represent numbers.
If not, might be better to convert it to `U256` before printing.

Reply:
The logic here is not changed, we tried to cast `H256` to `u32/u64` before and now only to `u64`,
both decimal and hexadecimal representations are printed.